### PR TITLE
Minting & Accept Terms

### PR DESCRIPTION
**Pull request by lupotto**
- `Accept Terms` button
- `Minting` button

Next: Test with an actual Minting contract which has the conditions stablished

### CODE REVIEW

File: pkg/examples/react/src/pages/index.tsx [hunk 1]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
   );
 };
 
+const AcceptTerms = () => {
+  const { account, active, library } = useWeb3React<Web3Provider>();
+  const [nftInputContractAddress, setNftInputContractAddress] = useState("");
+  //Redirecting to known contract that has accept terms activated
+  const nftContractAddress = nftInputContractAddress
+    ? ""
+    : "0x021aa5a885bdb6a8767e5e4909d1cf594255106a";
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:51]: can we not do this?
```suggestion
  const [nftInputContractAddress, setNftInputContractAddress] = useState("0x021aa5a885bdb6a8767e5e4909d1cf594255106a");
```

File: pkg/examples/react/src/pages/index.tsx [hunk 2]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+  };
+
+  async function getContractAgreement(
+    nftContractAddress: string,
+    signer: Signer
+  ): Promise<ICedarAgreementV0> {
+    return ICedarAgreementV0__factory.connect(nftContractAddress, signer);
+  }
+
+  async function tryGetContractAgreementIPFS(
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:88]:
```suggestion
  async function tryGetCedarAgreementContract(
```

File: pkg/examples/react/src/pages/index.tsx [hunk 3]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+    } catch (err) {
+      console.error(err);
+      return null;
+    }
+  }
+
+  return (
+    active && (
+      <div>
+        <text>Accept Terms Contract </text>
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:108]:
```suggestion
        <text>Contract Address</text>
```

File: pkg/examples/react/src/pages/index.tsx [hunk 4]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+    useWeb3React<Web3Provider>();
+
+  const onClick = async () => {
+    await getContract();
+  };
+  const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+  const ZERO_BYTES =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+  const getContract = async () => {
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:133]:
```suggestion
  const onMint = async () => {
```

File: pkg/examples/react/src/pages/index.tsx [hunk 5]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+    if (!library) return;
+
+    const contract = CedarERC721DropV0__factory.connect(
+      "0xd18d8a12c868EC243Afa3371c6C98Bf8631d7246",
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:137]: this should be the same address from the input

File: pkg/examples/react/src/pages/index.tsx [hunk 6]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+    });
+
+    const response = await contract.claim(account, 1, ZERO_ADDRESS, 0, [], 0, {
+      gasLimit: 100000,
+    });
+  };
+  return (
+    active && (
+      <div>
+        <button type="button" onClick={onClick}>
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:167]:
```suggestion
        <button type="button" onClick={onMint}>
```

File: pkg/examples/react/src/pages/index.tsx [hunk 7]
```diff
@@ -28,19 +35,151 @@ const ConnectWallet = () => {
...
+    )
+  );
+};
+
 const Home: NextPage = () => {
   return (
     <div>
       <main className={styles.main}>
-        <h2>Welcome to playground</h2>
+        <h2>Welcome to Aspen Publishing SDK </h2>
```
Comment (DeanVanNiekerk) [inline pkg/examples/react/src/pages/index.tsx:179]:
```suggestion
        <h2>Aspen Publishing SDK Example</h2>
```

**Review by DeanVanNiekerk — CHANGES_REQUESTED** [review]

**Comment by DeanVanNiekerk** [comment]
Mint and AcceptTerms are still potentially using different addresses, can we do something like

```tsx
import type { NextPage } from "next";
import styles from "../styles/Home.module.css";
import { useEffect, useState } from "react";

import { useWeb3React } from "@web3-react/core";
import { Web3Provider } from "@ethersproject/providers";
import { InjectedConnector } from "@web3-react/injected-connector";
import {
  CedarERC721DropV0__factory,
  ICedarAgreementV0,
  ICedarAgreementV0__factory,
} from "@monax/aspen-publishing-sdk";
import { Signer } from "ethers";

const ConnectWallet = () => {
  const injectedConnector = new InjectedConnector({
    supportedChainIds: [1, 3, 4, 5, 42, 80001],
  });
  const { chainId, account, activate, active, library } =
    useWeb3React<Web3Provider>();

  const onClick = () => {
    activate(injectedConnector);
  };

  useEffect(() => {
    console.log(chainId, account, active);
  });

  return (
    <div>
      <div>ChainId: {chainId}</div>
      <div>Account: {account}</div>
      {active ? (
        <div>✅ </div>
      ) : (
        <button type="button" onClick={onClick}>
          Connect
        </button>
      )}
    </div>
  );
};

const AcceptTerms: React.FC<{ contractAddress: string }> = ({
  contractAddress,
}) => {
  const { account, active, library } = useWeb3React<Web3Provider>();

  const onAcceptTerms = async () => {
    const contract = await getContractAgreement(
      contractAddress,
      library.getSigner()
    );
    try {
      await contract.termsActivated();
      await checkHasAcceptedTerms();
      await tryGetCedarAgreementContract(contractAddress, library.getSigner());

      await contract.acceptTerms();
    } catch (err) {
      console.error(err);
    }
  };

  const checkHasAcceptedTerms = async () => {
    if (!library) return;
    const erc721Cedar = await ICedarAgreementV0__factory.connect(
      contractAddress,
      library.getSigner()
    );
    return erc721Cedar.getAgreementStatus(account);
  };

  async function getContractAgreement(
    nftContractAddress: string,
    signer: Signer
  ): Promise<ICedarAgreementV0> {
    return ICedarAgreementV0__factory.connect(nftContractAddress, signer);
  }

  async function tryGetCedarAgreementContract(
    nftContractAddress: string,
    signer: Signer
  ): Promise<string | null> {
    try {
      const cedar = ICedarAgreementV0__factory.connect(
        nftContractAddress,
        signer
      );
      const agreementInIPFS = await cedar.userAgreement();
      return agreementInIPFS;
    } catch (err) {
      console.error(err);
      return null;
    }
  }

  return (
    active && (
      <button type="button" onClick={onAcceptTerms}>
        Accept Terms
      </button>
    )
  );
};

const Mint: React.FC<{ contractAddress: string }> = ({ contractAddress }) => {
  const { account, active, library } = useWeb3React<Web3Provider>();

  const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
  const ZERO_BYTES =
    "0x0000000000000000000000000000000000000000000000000000000000000000";

  const onMint = async () => {
    if (!library) return;
    const contract = CedarERC721DropV0__factory.connect(
      contractAddress,
      library.getSigner()
    );

    const condition = {
      maxClaimableSupply: 100,
      quantityLimitPerTransaction: 100,
      startTimestamp: Math.floor(Date.now() / 1000).toString(),
      supplyClaimed: 0,
      waitTimeInSecondsBetweenClaims: 0,
      merkleRoot: ZERO_BYTES,
      pricePerToken: 0,
      currency: ZERO_ADDRESS,
    };

    await contract.setClaimConditions([condition], false, {
      gasLimit: 100000,
    });

    await contract.lazyMint(1, "", ZERO_BYTES, {
      gasLimit: 100000,
    });

    const response = await contract.claim(account, 1, ZERO_ADDRESS, 0, [], 0, {
      gasLimit: 100000,
    });
  };
  return (
    active && (
      <div>
        <button type="button" onClick={onMint}>
          Mint
        </button>
      </div>
    )
  );
};

const Home: NextPage = () => {
  const [nftInputContractAddress, setNftInputContractAddress] = useState(
    "0x021aa5a885bdb6a8767e5e4909d1cf594255106a"
  );

  return (
    <div>
      <main className={styles.main}>
        <h2>Aspen Publishing SDK Example </h2>
        <ConnectWallet />
        <div>
          <text>Contract Address </text>
          <input
            type="text"
            value={nftInputContractAddress}
            onChange={(e) => setNftInputContractAddress(e.target.value)}
          />
        </div>
        <AcceptTerms contractAddress={nftInputContractAddress} />
        <Mint contractAddress={nftInputContractAddress} />
      </main>
    </div>
  );
};

export default Home;

```

File: pkg/examples/react/src/pages/index.tsx [hunk 8]
```diff
@@ -1,46 +1,265 @@
...
+  1: "Ethereum",
+  4: "Rinkeby",
+  137: "Polygon",
+  80001: "Mumbai",
+};
 
 const ConnectWallet = () => {
   const injectedConnector = new InjectedConnector({
-    supportedChainIds: [1, 3, 4, 5, 42],
+    supportedChainIds: [1, 4, 137, 80001],
```
Comment (danielz-mnx) [inline pkg/examples/react/src/pages/index.tsx:25]: hey @lupotto,
why not use the keys in `network` here? like:
```
supportedChainIds: Object.keys(network),
```

Comment (lupotto) [reply pkg/examples/react/src/pages/index.tsx:25]: yep good point! Thanks daniel!

**Review by DeanVanNiekerk — APPROVED** [review]

**Pull request merged**